<a href="https://colab.research.google.com/github/muhamadrefo/ImageClassificationColab/blob/master/NaiveBayesClassifier_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Prepare Environment

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/drive


In [2]:
#Importing Packages

import matplotlib.pyplot as plt
import csv
import numpy as np
import pandas as pd
#import seaborn as sns
#from datetime import datetime
#from matplotlib.colors import ListedColormap
#from pandas.plotting import scatter_matrix
#from sklearn.tree import DecisionTreeClassifier
#from sklearn.tree import export_graphviz
from sklearn import model_selection
from sklearn.model_selection import train_test_split as tt
from sklearn.metrics import classification_report
from sklearn.metrics import confusion_matrix
from sklearn.metrics import accuracy_score

%matplotlib inline

In [4]:
#Filtered Dataset
%cd /content/drive/My Drive/TestingPlace/PlaceofDatabase/CNN/
#Unfiltered Dataset
#%cd /content/drive/My Drive/Batik/

/content/drive/My Drive/TestingPlace/PlaceofDatabase/CNN


#Reading dataset

In [17]:
#col_names = ["Path", "Dissimilarity", "Contrast", "Homogeneity", "Energy", "Correlation", "ASM" , "Label"]
#datatrain = pd.read_csv('DataTraining.csv', names=col_names)

#data_train = pd.read_csv('DataTraining.csv') #DataTraining.csv nor DataValidation.csv
data_train = pd.read_csv('DataValidation.csv')
pd.set_option('display.max_columns', 100)
data_train.head(15) #display first 15 data

Path  Dissimilarity  Contrast  \
0   dataset_Cap_vs_Tulis/test/cap/YCNSBT.38.jpg       0.636491  0.934898   
1   dataset_Cap_vs_Tulis/test/cap/YCNSBT.31.jpg       0.552010  0.777266   
2   dataset_Cap_vs_Tulis/test/cap/YCNSBT.30.jpg       0.609825  0.833881   
3   dataset_Cap_vs_Tulis/test/cap/YCNSBT.27.jpg       0.610886  0.839266   
4   dataset_Cap_vs_Tulis/test/cap/YCNSBT.22.jpg       0.592649  0.845011   
5   dataset_Cap_vs_Tulis/test/cap/YCNSBT.29.jpg       0.618234  0.916760   
6   dataset_Cap_vs_Tulis/test/cap/YCNSBT.20.jpg       0.512532  0.691744   
7   dataset_Cap_vs_Tulis/test/cap/YCNSBT.13.jpg       0.501982  0.753984   
8   dataset_Cap_vs_Tulis/test/cap/YCNSBT.16.jpg       0.518618  0.725977   
9   dataset_Cap_vs_Tulis/test/cap/YCNSBT.15.jpg       0.437220  0.586803   
10  dataset_Cap_vs_Tulis/test/cap/YCNSBT.10.jpg       0.467369  0.628563   
11  dataset_Cap_vs_Tulis/test/cap/YCNSBT.18.jpg       0.637732  0.904829   
12  dataset_Cap_vs_Tulis/test/cap/YCNSBT.17.jpg       0.523202  0.786015   
13   dataset_Cap_vs_Tulis/test/cap/YCNSC.15.jpg       1.069106  2.316344   
14   dataset_Cap_vs_Tulis/test/cap/YCNSC.16.jpg       1.161715  2.514294   

    Homogeneity    Energy  Correlation       ASM Label  
0      0.711363  0.365774     0.466192  0.133791   cap  
1      0.746220  0.384418     0.539110  0.147778   cap  
2      0.717325  0.382543     0.404246  0.146339   cap  
3      0.717332  0.381105     0.425919  0.145241   cap  
4      0.728633  0.378711     0.450020  0.143422   cap  
5      0.719966  0.364927     0.450856  0.133172   cap  
6      0.761473  0.392889     0.573984  0.154362   cap  
7      0.773787  0.484037     0.585050  0.234292   cap  
8      0.761308  0.475221     0.589177  0.225835   cap  
9      0.796251  0.536811     0.571077  0.288166   cap  
10     0.782357  0.494227     0.613842  0.244261   cap  
11     0.707624  0.380243     0.390075  0.144585   cap  
12     0.764165  0.472486     0.541780  0.223243   cap  
13     0.585438  0.213968     0.628867  0.045782   cap  
14     0.549945  0.192834     0.596855  0.037185   cap

In [18]:
# show columns
data_train.columns 

Index(['Path', 'Dissimilarity', 'Contrast', 'Homogeneity', 'Energy',
       'Correlation', 'ASM', 'Label'],
      dtype='object')

# Encoding the Categorial Data

In [19]:
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
df = data_train.apply(le.fit_transform)
df.head(15)

Path  Dissimilarity  Contrast  Homogeneity  Energy  Correlation  ASM  \
0     12             96        93           29      35           21   35   
1     11             77        68           47      54           25   54   
2     10             90        79           31      50           10   50   
3      8             91        80           32      49           16   49   
4      7             87        81           36      47           18   47   
5      9             95        89           33      34           19   34   
6      6             64        56           58      55           33   55   
7      1             61        63           67      84           37   84   
8      3             67        59           57      83           39   83   
9      2             47        44           82     102           30  102   
10     0             50        49           73      87           52   87   
11     5             97        87           28      48            5   48   
12     4             70        70           60      80           26   80   
13    16            118       120            9       9           55    9   
14    17            126       127            3       5           42    5   

    Label  
0       0  
1       0  
2       0  
3       0  
4       0  
5       0  
6       0  
7       0  
8       0  
9       0  
10      0  
11      0  
12      0  
13      0  
14      0

# Splitting the Data into Training and Testing Data

In [20]:
X = df.iloc[:, 1:6].values
Y = df.iloc[:, 7].values
X_train, X_test, Y_train, Y_test = tt(X,Y, test_size=0.2, random_state=4)

In [21]:
X_train.shape

(106, 5)

In [22]:
Y.shape

(133,)

#Naive Bayes Model Start Here

In [23]:
from sklearn.naive_bayes import GaussianNB
nb = GaussianNB()
nb.fit(X_train,Y_train)
y_pred = nb.predict(X_test)

In [24]:
from sklearn.metrics import classification_report , confusion_matrix , accuracy_score
cm = confusion_matrix(Y_test,y_pred)

In [25]:
print("Confusion Matrix: \n",cm)

Confusion Matrix: 
 [[ 9 12]
 [ 0  6]]


In [26]:
print("Classification Report: \n", classification_report(Y_test, y_pred))

Classification Report: 
               precision    recall  f1-score   support

           0       1.00      0.43      0.60        21
           1       0.33      1.00      0.50         6

    accuracy                           0.56        27
   macro avg       0.67      0.71      0.55        27
weighted avg       0.85      0.56      0.58        27



In [27]:
print("Accuracy Score: ", accuracy_score(Y_test, y_pred))

Accuracy Score:  0.5555555555555556


# Analytic

using dataset unfiltered
Accuracy Score Training = 0.74
Accuracy Score Validation = 1.0

using dataset filtered
Accuracy Score Training = 0.77
Accuracy Score Validation = 0.55